gamemodes/terrortown/gamemode/roles/vindicator/vindicator.lua

```diff
@@ -90,7 +92,9 @@ hook.Add("PlayerDeath", "Vindicator_PlayerDeath", function(victim, infl, attacke
     local valid_kill = IsPlayer(attacker) and GetRoundState() == ROUND_ACTIVE
     if valid_kill then
         if victim:IsVindicator() and not victim:IsRoleActive() and attacker ~= victim then
-            if victim:IsZombifying() or attacker:IsHiveMind() then return end
+            print("Vindicator was killed!")
+            if attacker:IsVictimChangingRole(victim) then return end
+            print("They weren't changing role!")
```

Comment:
Should probably remove these
```suggestion
            if attacker:IsVictimChangingRole(victim) then return end
```